platforms/win32/vm/sqWin32Main.c

```diff
@@ -705,73 +727,45 @@ void gatherSystemInfo(void) {
 
   /* Find the driver key in the registry */
   keyName[0] = 0;
-  ok = RegOpenKey(HKEY_LOCAL_MACHINE,
-		  "HARDWARE\\DEVICEMAP\\VIDEO",
+  ok = RegOpenKeyW(HKEY_LOCAL_MACHINE,
+		  L"HARDWARE\\DEVICEMAP\\VIDEO",
 		  &hk);
   if (!ok) {
-    dwSize = 256;
-    RegQueryValueEx(hk,"\\Device\\Video0", NULL, NULL,
+    dwSize = 256*sizeof(WCHAR);
+    RegQueryValueExW(hk,L"\\Device\\Video0", NULL, NULL,
 		    (LPBYTE)keyName, &dwSize);
     RegCloseKey(hk);
   }
   if (*keyName) {
     /* Got the key name; open it and get the info out of there */
     char *tmp = tmpString + strlen(tmpString);
-    char deviceDesc[256];
-    char adapterString[256];
-    char biosString[256];
-    char chipType[256];
-    char dacType[256];
-    char *drivers, *drv;
+    char deviceDesc[MAX_PATH_UTF8];
+    char adapterString[MAX_PATH_UTF8];
+    char biosString[MAX_PATH_UTF8];
+    char chipType[MAX_PATH_UTF8];
+    char dacType[MAX_PATH_UTF8];
+    WCHAR *drivers, *drv;
     WCHAR buffer[256];
     DWORD memSize;
 
     /* Argh. It seems that the registry key regularly starts
        with \Registry\Machine\ which doesn't work with RegOpenKey below.
        I have no idea why but for now I'll just truncate that part if
        we recognize it... */
-    if (_strnicmp(keyName, "\\registry\\machine\\", 18) == 0) {
-      memcpy(keyName, keyName+18, strlen(keyName)-17);
+    if (_wcsnicmp(keyName, L"\\registry\\machine\\", 18) == 0) {
+      memmove(keyName, keyName+18*sizeof(WCHAR), (wcslen(keyName)-17)*sizeof(WCHAR));
```

Comment:
Err: `keyName` is `WCHAR*`, `keyName+18` skips 18 characters, `keyName+18*sizeof(WCHAR)` skips 36 characters!

Reply:
Fixed in 3e51616a56a0ef49b7c67a580db0f0fbaaea588f